database/models/fans.js

```diff
@@ -0,0 +1,132 @@
+const client = require('../index');
+
+module.exports = {
+  get: function(req, res) {
+    client.query(`
+      SELECT json_build_object(
+       'username', (SELECT username FROM authentication a WHERE a.id = f.authentication_id),
+       'address', (SELECT
+          json_build_object(
+            'city', f.city,
+            'state', f.state,
+            'street', f.street
+          )
+          FROM fans f
+          WHERE f.id = '${req.params.fanId}'
+        ),
+        'events', (SELECT
+          array_to_json(
+            array_agg(
+              json_build_object(
+                'event_id', e.id,
+                'event_name', e.name,
+                'event_street', e.street,
+                'event_city', e.city,
+                'event_state', e.state,
+                'event_longitude', e.longitude,
+                'event_latitude', e.latitude,
+                'event_timestamp', e.timestamp,
+                'event_start', e.start,
+                'event_end', e.end
+              )
+            )
+          )
+          FROM event_fan ef, events e
+          WHERE ef.events_id = e.id
+          AND ef.fans_id = '${req.params.fanId}'
+          GROUP BY ef.fans_id
+        ),
+        'artists', (SELECT
+          array_to_json(
+            array_agg(
+              json_build_object(
+                'artist_id', a.id,
+                'artist_display-name', a.display-name,
+                'artist_instrument', a.instrument,
+                'artist_genre', a.genre,
+                'artist_bio', a.bio,
+                'artist_picture', a.picture,
+                'artist_paymentmethod', a.paymentmethod,
+                'artist_fans_count', a.fans_count
+              )
+            )
+          )
+          FROM artist_fan af, artist a
+          WHERE af.artists_id = a.id
+          AND af.fans_id = '${req.params.fanId}'
+          GROUP BY af.fans_id
+        )
+      )
+      FROM fans f
+      WHERE f.id = '${req.params.fanId}'
+    `)
+    .then((data) => {
+      res.statusCode = 200;
+      res.end(JSON.stringify(data));
+    })
+    .catch((err) => {
+      res.status(500);
+      console.log('GET Fan Info error: ', err);
+      res.end(JSON.stringify(err));
+    });
+  },
+  saveEvent: function(req, res) {
+    client.query(`INSERT INTO event_fan
+                  (fans_id, events_id)
+                  VALUES
+                  (${req.body.id}, ${req.body.event_id})`)
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Save Event Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  followArtist: function(req, res) {
+    client.query(`INSERT INTO artist_fan
```

Comment:
artist_fan table is named as 'art_fan' in database / (fans_id, artists_id) are named as (art_id, fan_id) in art_fan table